waterloo-rocketry/minerva-rewrite
# Implement `/notify` command

## Description

Implemented the `/notify` command, which allows the user to either repost a message to specified channels or DM single-channels guests in those channels with the message. 



This PR closes #23 

## Developer Testing

Testing done:
- Implemented unit tests for `parseNotifyCommand()` and `parseEscapedSlashCommandChannel()`
- Manually tested edge-cases on the development slack

## Reviewer Testing



Steps to view/test Command
1. Copy the link to a message to use for the command
2. Run the `/notify` command in a channel with the copied link provided and `copy` argument passed (i.e. `/notify [link] copy`) and ensure that it is reposted to all default channels. Note that it will not repost to the channel that the message is in.
3. Run the `/notify` command with specific channels specified (i.e. `/notify [link] copy #general #test`) and ensure that it is reposted to all the specified channels.
4. Run the `/notify` command with the `copy-ping` argument specified (i.e. `/notify copy-ping [link] ping`) and ensure that the message is reposted and the messages ping `@channel`
5. Run the `/notify` command with just the link provided (i.e. `/notify [link]`) and ensure that any admins (stand-ins for single-channel guests) are DMd the message

Please also notify the feature reference document for this command: https://docs.google.com/document/d/1otYHtmFHzkN8g7089EGQpxi9-7C_iKFhDt2L2tuwxHs/edit#heading=h.8w6auka00s3p

- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/waterloo-rocketry/minerva-rewrite/69)
